src/h5web/visualizations/heatmap/HeatmapVis.tsx

```diff
@@ -30,24 +34,74 @@ function HeatmapVis(props: Props): JSX.Element {
     keepAspectRatio = true,
     showGrid = false,
     showLoader = true,
+    abscissas,
+    ordinates,
   } = props;
 
   const { rows, cols } = getDims(dataArray);
   const aspectRatio = keepAspectRatio ? cols / rows : undefined; // width / height <=> cols / rows
 
+  const abscissaValues = abscissas || range(cols + 1);
+  const abscissaToIndex = getValueToIndexScale(abscissaValues);
+  const abscissaDomain = useMemo(() => {
+    return getDataDomain(abscissaValues);
+  }, [abscissaValues]);
+
+  const ordinateValues = ordinates || range(rows + 1);
+  const ordinateToIndex = getValueToIndexScale(ordinateValues);
+  const ordinateDomain = useMemo(() => {
+    return getDataDomain(ordinateValues);
```

Comment:
Now I realise that this leads to a slight loss of performance. When `ordinates` (or `abscissas`) is not provided, we loop through the array to find the domain even though we know for a fact it's going to be `[0, rows + 1]`. Should we keep a ternary to avoid calling `getDataDomain`?

Also, should we rename `getDataDomain` and `useDataDomain` to just `getDomain` and `useDomain`?

Reply:
> Now I realise that this leads to a slight loss of performance. When `ordinates` (or `abscissas`) is not provided, we loop through the array to find the domain even though we know for a fact it's going to be `[0, rows + 1]`. Should we keep a ternary to avoid calling `getDataDomain`?

I was a bit concerned by this as well. I will look into it.

> Also, should we rename `getDataDomain` and `useDataDomain` to just `getDomain` and `useDomain`?

It would make more sense to me